src/libraries/System.Private.CoreLib/src/System/Reflection/CustomAttributeTypedArgument.cs

```diff
@@ -43,33 +43,47 @@ internal string ToString(bool typed)
                 return base.ToString()!;
 
             if (ArgumentType.IsEnum)
-                return string.Format(typed ? "{0}" : "({1}){0}", Value, ArgumentType.FullName);
-            else if (Value == null)
-                return string.Format(typed ? "null" : "({0})null", ArgumentType.Name);
-            else if (ArgumentType == typeof(string))
-                return string.Format("\"{0}\"", Value);
-            else if (ArgumentType == typeof(char))
-                return string.Format("'{0}'", Value);
-            else if (ArgumentType == typeof(Type))
-                return string.Format("typeof({0})", ((Type)Value!).FullName);
-            else if (ArgumentType.IsArray)
+                return typed ? $"{Value}" : $"({ArgumentType.FullName}){Value}";
+
+            if (Value == null)
+                return typed ? "null" : $"({ArgumentType.Name})null";
+
+            if (ArgumentType == typeof(string))
+                return $"\"{Value}\"";
+
+            if (ArgumentType == typeof(char))
+                return $"'{Value}'";
+
+            if (ArgumentType == typeof(Type))
+                return $"typeof({((Type)Value!).FullName})";
+
+            if (ArgumentType.IsArray)
             {
                 IList<CustomAttributeTypedArgument> array = (IList<CustomAttributeTypedArgument>)Value!;
-
                 Type elementType = ArgumentType.GetElementType()!;
-                string result = string.Format("new {0}[{1}] {{ ", elementType.IsEnum ? elementType.FullName : elementType.Name, array.Count);
+
+                var result = new ValueStringBuilder(stackalloc char[256]);
```

Comment:
Needs a `using` for `ValueStringBuilder`?

Reply:
Its ToString cleans up the state.